server/odc-service/src/main/java/com/oceanbase/odc/service/connection/database/DatabaseService.java

```diff
@@ -215,6 +215,29 @@ public Database detail(@NonNull Long id) {
         throw new NotFoundException(ResourceType.ODC_DATABASE, "id", id);
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    @SkipAuthorize("internal authenticated")
+    public List<Database> detailForMultipleDatabase(@NotEmpty List<Long> ids) {
+        List<DatabaseEntity> databaseEntities = databaseRepository.findAllById(ids);
+        List<Database> databases = databaseEntities.stream()
+                .map(entity -> entityToModel(entity, true))
+                .collect(Collectors.toList());
+        databases.forEach(database -> {
+            horizontalDataPermissionValidator.checkCurrentOrganization(database);
+            if (Objects.nonNull(database.getProject()) && Objects.nonNull(database.getProject().getId())) {
+                projectPermissionValidator.checkProjectRole(database.getProject().getId(), ResourceRoleName.all());
+            } else {
+                Permission requiredPermission = securityManager
```

Comment:
performance risk involved, may not judege each permission in loop, use batch check instead

Reply:
Restrict multiple databases to belong to the same project and organization, so the project and organization verification needs to be done only once